pallets/gear/src/schedule.rs

```diff
@@ -68,6 +68,11 @@ pub const STACK_HEIGHT_LIMIT: u32 = 36_743;
 #[cfg(feature = "fuzz")]
 pub const FUZZER_STACK_HEIGHT_LIMIT: u32 = 65_000;
 
+/// Maximum number of data segments in a wasm module.
+/// It has been determined that the maximum number of data segments in a wasm module
+/// does not exceed 1024 by a large margin.
+pub const DATA_SEGMENTS_AMOUNT_LIMIT: u32 = 1024;
```

Comment:
It has been determined that the maximum number of data segments in a wasm module
does not exceed 1024 by a large margin.

See https://github.com/gear-tech/gear/issues/3907#issuecomment-2083556107